### PR TITLE
Linting: fix ignoring files_unchanged

### PR DESCRIPTION
**Pull request by mirpedrol**

### CODE REVIEW

**Review by mashehu — APPROVED** [review]

File: nf_core/lint/files_exist.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import logging
 from pathlib import Path
-from typing import Union
+from typing import Dict, List, Tuple, Union
+
+from nf_core.lint import PipelineLint
```
Comment (mashehu) [inline nf_core/lint/files_exist.py:5]:
```suggestion
```

File: nf_core/lint/files_exist.py [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 
 log = logging.getLogger(__name__)
 
 
-def files_exist(self):
+def files_exist(self: PipelineLint) -> dict[str, Union[List[str], bool]]:
```
Comment (mashehu) [inline nf_core/lint/files_exist.py:10]:
```suggestion
def files_exist(self) -> dict[str, Union[List[str], bool]]:
```

File: nf_core/lint/files_unchanged.py [hunk 3]
```diff
@@ -3,16 +3,17 @@
 import shutil
 import tempfile
 from pathlib import Path
-from typing import Union
+from typing import Dict, List, Tuple, Union
 
 import yaml
 
 import nf_core.create
+from nf_core.lint import PipelineLint
```
Comment (mashehu) [inline nf_core/lint/files_unchanged.py:11]:
```suggestion
```

File: nf_core/lint/files_unchanged.py [hunk 4]
```diff
@@ -3,16 +3,17 @@
...
 
 log = logging.getLogger(__name__)
 
 
-def files_unchanged(self):
+def files_unchanged(self: PipelineLint) -> dict[str, Union[List[str], bool]]:
```
Comment (mashehu) [inline nf_core/lint/files_unchanged.py:16]:
```suggestion
def files_unchanged(self) -> dict[str, Union[List[str], bool]]:
```

**Comment by mashehu** [comment]
@nf-core-bot fix linting

**Comment by mirpedrol** [comment]
@nf-core-bot fix linting

File: nf_core/lint/files_exist.py [hunk 5]
```diff
@@ -129,19 +129,19 @@ def files_exist(self):
...
+        [Path(".gitattributes")],
+        [Path(".gitignore")],
+        [Path(".nf-core.yml")],
+        [Path(".editorconfig")],
+        [Path(".prettierignore")],
+        [Path(".prettierrc.yml")],
+        [Path("CHANGELOG.md")],
+        [Path("CITATIONS.md")],
+        [Path("CODE_OF_CONDUCT.md")],
+        [Path("LICENSE", "LICENSE.md", "LICENCE", "LICENCE.md")],  # NB: British / American spelling
```
Comment (mashehu) [inline nf_core/lint/files_exist.py:141]:
```suggestion
        [Path("LICENSE"), Path("LICENSE.md"), Path("LICENCE"), Path("LICENCE.md")],  # NB: British / American spelling
```

File: nf_core/lint/files_exist.py [hunk 6]
```diff
@@ -242,22 +244,24 @@ def pf(file_path: Union[str, Path]) -> Path:
...
+            failed.append(f"File must be removed: {self._wrap_quotes(file_cond[0])}")
+        elif pf(file_cond[0]).is_file() and not in_config:
+            passed.append(f"File found check: {self._wrap_quotes(file_cond[0])}")
+        elif not pf(file_cond[0]).is_file() and not in_config:
+            failed.append(f"File not found check: {self._wrap_quotes(file_cond[0])}")
+        elif not pf(file_cond[0]).is_file() and in_config:
+            passed.append(f"File not found check: {self._wrap_quotes(file_cond[0])}")
     # Files that cause a warning if they exist
     for file in files_warn_ifexists:
         if file in ignore_files:
```
Comment (mirpedrol) [inline nf_core/lint/files_exist.py:267]:
```suggestion
        if str(file) in ignore_files:
```

Comment (mashehu) [reply nf_core/lint/files_exist.py:267]: ignore_files are already be strings I think

Comment (mirpedrol) [reply nf_core/lint/files_exist.py]: The files in `files_warn_ifexists` are paths, so I think we have to convert them to strings to compare with the ones in `ignore_files`

File: nf_core/lint/files_unchanged.py [hunk 7]
```diff
@@ -215,37 +215,34 @@ def _tf(file_path: Union[str, Path]) -> Path:
...
-    for files in files_conditional:
+    for file_conditional in files_conditional:
         # Ignore if file specified in linting config
         ignore_files = self.lint_config.get("files_unchanged", [])
-        if files[0] in ignore_files:
-            ignored.append(f"File ignored due to lint config: {self._wrap_quotes(files)}")
-
-        # Ignore if we can't find the file
-        elif _pf(files[0]).is_file():
-            ignored.append(f"File does not exist: {self._wrap_quotes(files[0])}")
```
Comment (mashehu) [inline nf_core/lint/files_unchanged.py:226]: Not happy about having to drop this, but the only way to make lint with `fail-ignore` happy

File: tests/test_lint.py [hunk 8]
```diff
@@ -200,6 +200,7 @@ def test_sphinx_md_files(self):
     )
     from .lint.files_exist import (  # type: ignore[misc]
         test_files_exist_depreciated_file,
+        test_files_exist_fail_conditional,
```
Comment (mirpedrol) [inline tests/test_lint.py:203]:
```suggestion
        test_files_exist_fail_conditional,
        test_files_exist_pass_conditional,
```

**Comment by mashehu** [comment]
@nf-core-bot fix linting

**Pull request merged**